design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,188 @@
+
+The deployment file for the operator and example of CRD instance will be included in every project release.
+
+The ClusterLink CLI will be utilized to automate the deployment process.
+For instance, the following commands can assist the fabric administrator in creating the site certificates:
+
+    clusterlink create fabric --name <fabric_name>
+    clusterlink create site --name <site_name> --fabric <fabric_name> 
+
+For the site administrator, automation of deploying the certificates as a secret to the cluster, deploying the ClusterLink operator, and the CRD instance can be achieved with:
+
+    clusterlink install --site <site_name>
+
+Note: The ClusterLink CLI uses the kubeconfig, and it should be set to the specific cluster that needs to be configured
+
+### ClusterLink CRD
+
+The ClusterLink CRD includes the following fields:
+
+* **API version:** clusterlink.net/v1alpha1
+* **Kind:** clusterlink
+
+* **Spec:**
+  
+    | Field name | Subfield name| Description | Default value |
+    | ---- | ----- | ------ | ----|
+    |dataplane | | ClusterLink dataplane component attributes||
+    |  |type| Types of dataplane, supported values "go" or "envoy"|"envoy"|
+    | |replicas| Number of dataplane replicas|1|
+    |ingress|| ClusterLink ingress component attributes ||
+    ||type| Type of service to expose ClusterLink deployment, supported values: "LoadBalancer", "Gateway", "NodePort", "None". |None|
+    ||Port| Port represents the port number of the external service |443 for all types,except for NodePort, where the port number will be allocated by Kubernetes | 
+    |logLevel| |Log level severity for all the components (controlplane and dataplane)| "info"|
+    |containerRegistry| |The container registry to pull the project images when the images is not present locally | ghcr.io/clusterlink-net|
+    |imageTag| |The project images version | latest|
+    |Namespace| | The namespace where the components of the ClusterLink project are deployed | clusterlink-system|
+  .
+
+* **Status:**
+
+    | Field name | Subfield name|  Description |
+    | ----------- | ----------- |----------- |
+    | controlplane || Status of the controlplane components controlled by the operator |
+    |  |Conditions|The controlplane will have two status conditions: 1. ```DeploymentReady``` will be set to true when the controlplane deployment is ready, and 2. `ServiceReady` will be set to true when the controlplane service is ready|
+    | datalplane || Status of the dataplane components controlled by the operator |
+    |  |Conditions|The dataplane will have two status conditions: 1. ```DeploymentReady``` will be set to true when the dataplane deployment is ready, and 2. `ServiceReady` will be set to true when the dataplane service is ready|
+    | ingress || Status of the dataplane components controlled by the operator |
+    |  |Conditions|The ingress will have one status conditions: `ServiceReady` will be set to true when the external ingress service is ready|
```

Comment:
You may want to consider two more conditions (along with an observed generation value):
- `Reconciling`: set to True when the controller is actively working; and
- `Stalled`: when some error condition has occurred (e.g., failed call to API master, secondary CRD instances being ignored, etc.)